packages/fxa-settings/src/lib/relier-factory.ts

```diff
@@ -0,0 +1,176 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Context } from "./context/interfaces/context";
+import { RelierData, Relier } from "../models/reliers/rellier"
+import { BrowserRelier } from "../models/reliers/browser-relier";
+import { OAuthRelier } from "../models/reliers/oauth-relier";
+import { RelierPairingSupplicant } from "../models/reliers/pairing-supplicant-relier";
+import { PairingAuthorityRelier } from "../models/reliers/pairing-authority-relier";
+import { UrlContext } from "./context/implementations/url-context";
+import { StorageContext } from "./context/implementations/storage-context";
+
+const DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP = /^\/pair\/supp/;
+const DEVICE_PAIRING_AUTHORITY_REDIRECT_URI = 'urn:ietf:wg:oauth:2.0:oob:pair-auth-webchannel';
+const SYNC_SERVICE = 'sync';
```

Comment:
I reference the `MozServices` enum further down in review, this makes me wonder where we should account for 'sync' vs 'Firefox Sync' hmm.

Reply:
yeah... I am sure there are kinks with our state to work out. Hopefully testing shines some light on this. Also maybe these consts should go somewhere else?

Reply:
I'll have to file a follow up for this. I tried to change this, but once I made the switch, there were quite a few places that broke. It seems like we have a handful of hard coded strings that are anticipated or even set as defaults which don't align with MozServices. See FXA-6802.